packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,1130 @@
+    let {
+      className,
+      label,
+      options,
+      disabled,
+      ariaLabel,
+      required,
+      errorMessage,
+      onRenderContainer = this._onRenderContainer,
+      allowFreeform,
+      autoComplete,
+      buttonIconProps
+    } = this.props;
+    let { isOpen, selectedIndex, focused, suggestedDisplayValue, currentOptions } = this.state;
+    let selectedOption = currentOptions[selectedIndex];
+
+    return (
+      <div ref='root' className={ css('ms-ComboBox-container') }>
+        { label && (
+          <Label className={ css('ms-ComboBox-label') } id={ id + '-label' } ref={ this._resolveRef('_comboBoxLabel') } required={ required } htmlFor={ id }>{ label }</Label>
+        ) }
+        <div
+          ref={ this._resolveRef('_comboBoxWrapper') }
+          id={ id + 'wrapper' }
+          className={
+            css('ms-ComboBox', styles.wrapper,
+              (errorMessage && errorMessage.length > 0 ? styles.wrapperForError : null),
+              styles.root, className, {
+                'is-open': isOpen,
+                ['is-disabled ' + styles.rootIsDisabled]: disabled,
+                'is-required ': required,
+                [styles.focused]: focused,
+                [styles.readOnly]: !allowFreeform
+              }
+            )
+          } >
+          <DynamicAutoFill
+            data-is-interactable={ !disabled }
+            ref={ this._resolveRef('_comboBox') }
+            id={ id + '-input' }
+            className={ styles.input }
+            type='text'
+            key={ selectedIndex }
+            onFocus={ this._select }
+            onBlur={ this._onBlur }
+            onKeyDown={ this._onInputKeyDown }
+            onKeyUp={ this._onInputKeyUp }
+            onClick={ allowFreeform ? this.focus : this._onComboBoxClick }
+            onInputValueChange={ this._onInputChange }
+            aria-expanded={ isOpen }
+            aria-autocomplete={ (!disabled && autoComplete) }
+            role='combobox'
+            aria-readonly={ ((allowFreeform || disabled) ? null : 'true') }
+            readOnly={ disabled || !allowFreeform }
+            aria-labelledby={ (label && (id + '-label')) }
+            aria-label={ ((ariaLabel && !label) && ariaLabel) }
+            aria-describedby={ (id + '-option') }
+            aria-activedescendant={ (isOpen && selectedIndex >= 0 ? (id + '-list' + selectedIndex) : null) }
+            aria-disabled={ disabled }
+            aria-owns={ (id + '-list') }
+            defaultVisibleValue={ this._getVisibleValue() }
+            suggestedDisplayValue={ suggestedDisplayValue } />
+          <IconButton
+            className={ styles.caretDown }
+            role='presentation'
+            aria-hidden='true'
+            tabIndex={ -1 }
+            onClick={ this._onComboBoxClick }
+            iconProps={ buttonIconProps } />
+        </div>
+
+        { isOpen && (
+          onRenderContainer(this.props, this._onRenderContainer)
```

Comment:
I followed the way other components are doing things like this, but I've updated all the places where we were passing the props to a callback